app/models/grouping.rb

```diff
@@ -669,24 +669,31 @@ def self.get_groupings_for_assignment(assignment, user)
                 .where(user: user)
                 .select { |m| m.grouping.is_valid? }
                 .map &:grouping
+    elsif user.is_a_reviewer?(assignment)
+      # grab only the groupings of reviewees that this reviewer
+      # is responsible for
+      user_group = user.grouping_for(assignment.id)
+      groupings = user_group.peer_reviews_to_others
+      groupings.map {|p| Result.find(p.result_id).submission.grouping}
     else
-      assignment.groupings.joins(:memberships)
-              .includes(:assignment,
-                        :group,
-                        :grace_period_deductions,
-                        :tags,
-                        { current_submission_used: [:results,
-                                                    :submission_files,
-                                                    :submitted_remark,
-                                                    grouping: :group] },
-                        { accepted_student_memberships: :user },
-                        { inviter: :section }
-                        )
-              .where(memberships: { membership_status:
-                                   [StudentMembership::STATUSES[:inviter],
-                                    StudentMembership::STATUSES[:pending],
-                                    StudentMembership::STATUSES[:accepted]] })
-              .distinct
+      groupings = assignment.groupings
```

Comment:
If you mean the extra `elsif` block, it's because I just had to filter out the groupings that the reviewer is responsible for, instead of all the assignments groupings.